opennms-webapp-rest/src/main/java/org/opennms/web/rest/v1/ResourceRestService.java

```diff
@@ -123,4 +138,21 @@ public ResourceDTO getResourceForNode(@PathParam("nodeCriteria") final String no
 
         return ResourceDTO.fromResource(resource, depth);
     }
+
+
+    @POST
+    @Path("generateId")
+    @Consumes({MediaType.APPLICATION_JSON})
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response saveResourcesWithId(String[] resources) {
+        if (m_jsonStore == null) {
+            throw getException(Status.NOT_FOUND, "No JsonStore registered.");
+        }
+        String resourcesInJson = new Gson().toJson(resources);
```

Comment:
Create a private final member in this bean that contains the `new Gson()` instance and reuse it here rather than create a new object each `POST`

```suggestion
        String resourcesInJson = m_gson.toJson(resources);
```